### PR TITLE
Make Extent GDAL parameter optional

### PR DESCRIPTION
**Pull request by strk**
Use "auto" to have it automatically computed to min covering extent.
See  http://hub.qgis.org/issues/15685

### CODE REVIEW

File: python/plugins/processing/gui/ExtentSelectionPanel.py [hunk 1]
```diff
@@ -205,10 +205,10 @@ def setValueFromRect(self, r):
         self.dialog.activateWindow()
 
     def getValue(self):
-        if unicode(self.leText.text()).strip() != '':
-            return unicode(self.leText.text())
-        else:
+        if unicode(self.leText.text()).strip() == 'auto':
             return self.getMinCoveringExtent()
+        else:
+            return unicode(self.leText.text())
```
Comment (m-kuhn) [inline python/plugins/processing/gui/ExtentSelectionPanel.py]: Is it required to cast this explicitly to unicode?
Skipping explicit unicode casts wherever possible will make code more portable between python 2/3.

Comment (strk) [inline python/plugins/processing/gui/ExtentSelectionPanel.py]: No idea, I just left that part untouched. @volaya  ?

**Comment by strk** [comment]
@gioman I could warp with the code in this PR

**Comment by gioman** [comment]
> @gioman I could warp with the code in this PR

@strk I will test asap and let you know

**Comment by gioman** [comment]
> @gioman I could warp with the code in this PR

@strk if the "raster extent..." is left blank than now it works (it is not automatically used the "te" parameter filled with the full extent of the input) BUT if you fill that parameter with an extent then


2016-10-19T17:38:21	2	Uncaught error while executing algorithm
			Traceback (most recent call last):
			  File "C:/OSGeo4W/apps/qgis-dev/./python/plugins\processing\core\GeoAlgorithm.py", line 203, in execute
			    self.processAlgorithm(progress)
			  File "C:/OSGeo4W/apps/qgis-dev/./python/plugins\processing\algs\gdal\GdalAlgorithm.py", line 51, in processAlgorithm
			    commands = self.getConsoleCommands()
			  File "C:/OSGeo4W/apps/qgis-dev/./python/plugins\processing\algs\gdal\warp.py", line 203, in getConsoleCommands
			    return ['gdalwarp', GdalUtils.escapeAndJoin(arguments)]
			  File "C:/OSGeo4W/apps/qgis-dev/./python/plugins\processing\algs\gdal\GdalUtils.py", line 174, in escapeAndJoin
			    if s[0] != '-' and ' ' in s:
			IndexError: string index out of range

**Comment by gioman** [comment]
@strk @volaya @alexbruy there is also the need to understand what to do when QGIS is compiled against gdal 1.1*, see the cases I raised here

http://hub.qgis.org/issues/15685

**Comment by strk** [comment]
On Wed, Oct 19, 2016 at 09:40:43AM -0700, Giovanni Manghi wrote:
> 			  File "C:/OSGeo4W/apps/qgis-dev/./python/plugins\processing\algs\gdal\GdalUtils.py", line 174, in escapeAndJoin
> 			    if s[0] != '-' and ' ' in s:
> 			IndexError: string index out of range

Should be fixed with 533ca7c38d46500a6d234342da3745de5e304f6e
Can you confirm ? (it was about a now optional extent CRS)

**Comment by gioman** [comment]
> Can you confirm ? (it was about a now optional extent CRS)

yes it works. Anyway there is a thing that if possible should be addressed:

imagine you have raster A with CRS 123, QGIS project has OTF ON and CRS 123, and you want to reproject it to CRS 456

when filling the extent for the "te" parameter, then IF the "te_srs" parameter is NOT specified (in this case should be 123) then GDALWARP assumes that is 456 (target CRS), and the result is... wrong/empty.

Similar scenario, but project CRS is 456 now. The extent for "te" is defined, "te_srs" it NOT specified but now the result is correct, because gdalwap assumes correctly the the coordinates are in the target CRS (456).

Long story short: in processing would be needed to make the parameter "CRS of raster extent" (only) when "raster extent" is filled.

**Comment by strk** [comment]
On Wed, Oct 19, 2016 at 11:17:36AM -0700, Giovanni Manghi wrote:

> Long story short: in processing would be needed to make the parameter "CRS of raster extent" (only) when "raster extent" is filled.

To make the parameter _mandatory_, you mean ?

**Comment by gioman** [comment]
> To make the parameter _mandatory_, you mean ?

yes, I edited the comment.

**Comment by strk** [comment]
Since ogr2ogr does have a default for the extent CRS (target CRS) why making it mandatory to specify ?
With d9c0aa67bd45640574d92141fc26cbe3559cafec I tweaked the label to make it documented

**Comment by strk** [comment]
Well, pushed as dc2df62b88cdc01b0fc8839779da44fb87ccd213

**Comment by gioman** [comment]
> Since ogr2ogr does have a default for the extent CRS (target CRS) why making it mandatory to specify ?

users do not (usually) fill the "extent of raster..." manually, they use the Processing functionality to drag the mouse on canvas to draw a bbox, and the resulting values are always in project CRS, that could be very well NOT the target CRS. Is this anyway a detail, not even sure if now Processing allows to have parameters mandatory only when another parameter has been filled.

**Comment by gioman** [comment]
> With d9c0aa6 I tweaked the label to make it documented

it helps. Anyway the ticket should be left open unless also the issues with the tool when gdal < 2.* is used have been also fixed. Is the case?

**Comment by strk** [comment]
I hadn't tested with GDAL < 2.*
Please file a new ticket if it's broken
